mx_fetchjdk.py

```diff
@@ -0,0 +1,147 @@
+from jdk_distribution_parser import JdkDistribution
+import mx, mx_urlrewrites
+
+@command('mx', 'fetch-jdk', '[options]')
+@optional_suite_context
+def fetch_jdk(args):
+    """fetches required JDK version
+
+    If called without --quiet flag, menu will be printed for available JDK selection.
+    """
+    args = _parse_fetchjdk_settings(args)
+
+    distribution = args["java_distribution"]
+    jdk_path = args["jdk_path"]
+    jdk_folder = distribution.get_jdk_folder()
+    full_jdk_path = distribution.get_full_jdk_path(jdk_path)
+    jdk_url = mx_urlrewrites.rewriteurl(distribution.get_url())
+    jdk_archive = distribution.get_archive()
+
+    if not exists(full_jdk_path):
+        mx._opts.no_download_progress = args["quiet"]
+        mx.download(join(jdk_path, jdk_archive), [jdk_url], verbose=not args["quiet"])
+        untar = mx.TarExtractor(join(jdk_path, jdk_archive))
+
+        if not args["quiet"]:
+            print("Installing...")
+
+        with untar._open() as tar_file:
+            curr_full_jdk_path = untar._getnames(tar_file)[0]
+
+        untar.extract(jdk_path)
+        if not args["keep-archive"]:
+            os.remove(join(jdk_path, jdk_archive))
+        os.rename(join(jdk_path, curr_full_jdk_path), full_jdk_path)
+
+    if mx.get_os() == 'darwin':
+        full_jdk_path = join(full_jdk_path, 'Contents', 'Home')
+
+    if "alias" in args:
+        alias_full_path = join(jdk_path, args["alias"])
+        if exists(alias_full_path) or islink(alias_full_path):
+            if isdir(alias_full_path) and not islink(alias_full_path):
+                shutil.rmtree(alias_full_path)
+            else:
+                os.remove(alias_full_path)
+
+        if not (mx.is_windows() or mx.is_cygwin()):
+            os.symlink(jdk_folder, alias_full_path)
+        else:
+            copytree(full_jdk_path, alias_full_path, symlinks=True) # fallback for windows
+        full_jdk_path = alias_full_path
+
+    if not args["quiet"]:
+        print("Run the following to set JAVA_HOME in your shell:")
+
+    print('export JAVA_HOME={}'.format(abspath(full_jdk_path)))
+
+
+def _parse_fetchjdk_settings(args):
+    parser = ArgumentParser(prog='mx fetch-labsjdk')
+    parser.add_argument('--java-distribution', action='store', help='JDK distribution which should be downloaded')
+    parser.add_argument('--configuration', action='store', help='location of configuration json file')
+    parser.add_argument('--to', action='store', help='location where JDK would be downloaded')
+    parser.add_argument('--alias', action='store', help='name of symlink to JDK')
+    parser.add_argument('--keep-archive', action='store_true', help='keep downloaded JDK archive')
+    parser.add_argument('-q', '--quiet', action='store_true', help='suppress logging output')
+    parser.add_argument('remainder', nargs=REMAINDER, metavar='...')
+    args = parser.parse_args(args)
+
+    settings = {}
+    settings["jdk_path"] = join('bin', 'jdks')
```

Comment:
This should be `os.path.abspath(join('bin', 'jdks'))`.

Reply:
Fixed